File: app/src/main/java/com/agrogames/islandsofwar/engine/impl/game/BulletAdder.java
package com.agrogames.islandsofwar.engine.impl.game;

import com.agrogames.islandsofwar.engine.abs.game.Bullet;

import java.util.ArrayList;
import java.util.List;

public class BulletAdder implements com.agrogames.islandsofwar.engine.abs.game.BulletAdder {
    private final List<Bullet> bullets = new ArrayList<>();
    @Override
    public void AddBullet(Bullet bullet) {
        bullets.add(bullet);
    }

    public List<Bullet> getBullets(){
        return bullets;
    }
}

File: app/src/main/java/ru/agrogames/islands/engine/impl/bullet/BulletAdder.kt
package ru.agrogames.islands.engine.impl.bullet

import ru.agrogames.islands.engine.abs.bullet.BulletAdder
import ru.agrogames.islands.engine.abs.bullet.IBullet
import ru.agrogames.islands.sounds.SoundPlayer

class BulletAdder() : BulletAdder {
    override val bullets: MutableList<IBullet> = ArrayList()
    override fun addBullet(bullet: IBullet) {
        bullets.add(bullet)
        if (bullet.creationSound != null) {
            SoundPlayer.playSound(bullet.creationSound)
        }
    }
}